lib/validation.js

```diff
@@ -29,6 +30,18 @@ function build (opts, compile) {
     return
   }
 
+  if (opts.schema.headers) {
+    // headers will always be an object, allow schema def to skip this
+    if (!opts.schema.headers.type || !opts.schema.headers.properties) {
```

Comment:
Why?

Reply:
Because we also provide it for querystring

Reply:
My bad,